src/DIRAC/Core/Utilities/DIRACScript.py

```diff
@@ -0,0 +1,126 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+__RCSID__ = "$Id$"
+
+from collections import defaultdict
+import functools
+
+import six
+
+from DIRAC import gLogger
+
+
+class DIRACScript(object):
+  """Decorator for providing command line executables
+
+  All console-scripts entrypoints in DIRAC and downstream extensions should be
+  wrapped in this decorator to allow extensions to override any entry_point.
+  """
+  def __init__(self):
```

Comment:
The fact this decorator is a class is used to detect which script name is currently being ran:

https://github.com/DIRACGrid/DIRAC/blob/27ca8d8fb1c2a052828e149c894c739002445786/src/DIRAC/Core/Utilities/DIRACScript.py#L72-L74

It could be done by looking at `sys.argv` and the current executable but that's much fiddlier to do correctly. Actually to make debugging easier I could add a `if not isinstance(self, DIRACScript):` check to compliment the `ImportError` that ensures all scripts have been wrapper correctly:

https://github.com/DIRACGrid/DIRAC/blob/27ca8d8fb1c2a052828e149c894c739002445786/src/DIRAC/Core/Utilities/DIRACScript.py#L62-L70

For the docs, they will definitely be inclued in this PR. I just want to make sure we've settled on the implementation first before I waste time .